pkg/crc/machine/machine.go

```diff
@@ -765,29 +765,32 @@ func getProxyConfig(baseDomainName string) (*network.ProxyConfig, error) {
 
 func configProxyForCluster(ocConfig oc.OcConfig, sshRunner *crcssh.SSHRunner, sd *systemd.InstanceSystemdCommander,
 	proxy *network.ProxyConfig, instanceIP string) (err error) {
-	if proxy.IsEnabled() {
-		defer func() {
-			// Restart the crio service
-			if proxy.IsEnabled() {
-				// Restart reload the daemon and then restart the service
-				// So no need to explicit reload the daemon.
-				if _, ferr := sd.Restart("crio"); ferr != nil {
-					err = ferr
-				}
-			}
-		}()
-		proxy.ApplyToEnvironment()
+	if !proxy.IsEnabled() {
+		return nil
+	}
 
-		logging.Info("Adding proxy configuration to the cluster ...")
-		proxy.AddNoProxy(instanceIP)
-		if err := cluster.AddProxyConfigToCluster(ocConfig, proxy); err != nil {
-			return err
+	defer func() {
+		// Restart the crio service
+		if proxy.IsEnabled() {
+			// Restart reload the daemon and then restart the service
+			// So no need to explicit reload the daemon.
+			if _, ferr := sd.Restart("crio"); ferr != nil {
+				err = ferr
+			}
 		}
+	}()
+	proxy.ApplyToEnvironment()
```

Comment:
@cfergeau why we again have to apply to env?

Reply:
This `proxy.ApplyToEnvironment()` call was already there before this PR, it could indeed be removed by the first commit, I'll do that.